File: scripts/clean.lua
local directories = require("./directories.lua")
local prelude = require("./prelude")

local target = arg[1]
local valid_targets = directories.directories


local function clean_console()
    os.execute("cd console && rm -rf node_modules dist .nuxt && cd src-tauri && cargo clean")
end

local function clean_kernel()
    os.execute("cd kernel && cargo clean")
end

local function clean_server()
    os.execute("cd server && cargo clean")
end

local function clean()
    if not prelude.contains(valid_targets, target) then
        print("Invalid target '" .. (target or "") .. "'. Use one of: console, kernel, server, all")
        os.exit(1)
    end

    print("Cleaning " .. target .. " build assets")

    if target == "console" then
        clean_console()
    elseif target == "kernel" then
        clean_kernel()
    elseif target == "server" then
        clean_server()
    elseif target == "all" then
        clean_console()
        clean_kernel()
        clean_server()
    end
end

clean()

File: scripts/clean.mjs
#!/usr/bin/env node
import { execSync } from "node:child_process";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";
import { DIRECTORIES } from "./directories.mjs";
import { contains } from "./prelude.mjs";

const ROOT = resolve(dirname(fileURLToPath(import.meta.url)), "..");

function cleanConsole() {
  execSync("rm -rf node_modules dist .nuxt", {
    cwd: resolve(ROOT, "console"),
    stdio: "inherit",
  });
  execSync("cargo clean", {
    cwd: resolve(ROOT, "console/src-tauri"),
    stdio: "inherit",
  });
}

function cleanServer() {
  execSync("cargo clean", { cwd: resolve(ROOT, "server"), stdio: "inherit" });
}

function clean() {
  const target = process.argv[2];

  if (!contains(DIRECTORIES, target)) {
    console.error(
      `Invalid target '${target ?? ""}'. Use one of: console, server, all`,
    );
    process.exit(1);
  }

  console.log(`Cleaning ${target} build assets`);

  try {
    if (target === "console") {
      cleanConsole();
    } else if (target === "server") {
      cleanServer();
    } else if (target === "all") {
      cleanConsole();
      cleanServer();
    }
  } catch (err) {
    console.error(err.stderr ?? err.message);
    process.exit(1);
  }
}

clean();
